presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchange.java

```diff
@@ -79,7 +90,7 @@ public LocalExchange(
             int sinkFactoryCount,
             int bufferCount,
             PartitioningHandle partitioning,
-            List<? extends Type> types,
+            Optional<PartitionFunction> partitionFunction,
```

Comment:
Add comment stating `partitionFunction` should only present when `partitioning` represents a partitioned exchange?

I am also thinking if it makes sense to compute the `partitionFunction` inside this method? Note here is contract from callee that `partitionFunction` has to be consistent with `partitioning`.

Reply:
> I am also thinking if it makes sense to compute the partitionFunction inside this method?

Actually yes =)